scalajs/src/main/scala/clue/js/WebSocketJSBackend.scala

```diff
@@ -34,61 +35,60 @@ final class WebSocketJSBackend[F[_]: Async: Logger] extends WebSocketBackend[F]
-                case str: String => handler.onMessage(connectionId, str)
-                case other       => s"Unexpected event from WebSocket for [$uri]: [$other]".errorF
-              }
-              dispatcher.unsafeRunAndForget(message)
+          ws.onmessage = { e: MessageEvent =>
+            val message: F[Unit] = e.data match {
+              case str: String => handler.onMessage(connectionId, str)
+              case other       => s"Unexpected event from WebSocket for [$uri]: [$other]".errorF
             }
+            dispatcher.unsafeRunAndForget(message)
+          }
 
-            // According to spec, onError is only closed prior to a close.
-            // https://html.spec.whatwg.org/multipage/web-sockets.html
-            ws.onerror = { _: Event =>
-              val error: F[Unit] = (
-                for {
-                  _    <- s"Error on WebSocket for [$uri]".errorF
-                  _    <- isErrored.set(true)
-                  open <- isOpen.get
-                } yield if (!open) cb(new ConnectionException().asLeft)
-              ).uncancelable
-              dispatcher.unsafeRunAndForget(error)
-            }
+          // According to spec, onError is only closed prior to a close.
+          // https://html.spec.whatwg.org/multipage/web-sockets.html
+          ws.onerror = { _: Event =>
+            val error: F[Unit] = (
+              for {
+                _    <- s"Error on WebSocket for [$uri]".errorF
+                _    <- isErrored.set(true)
+                open <- isOpen.get
+              } yield if (!open) cb(new ConnectionException().asLeft)
+            ).uncancelable
+            dispatcher.unsafeRunAndForget(error)
+          }
 
-            ws.onclose = { e: CloseEvent =>
-              val close: F[Unit] =
-                for {
-                  _       <- s"WebSocket closed for URI [$uri]".traceF
-                  errored <- isErrored.get
-                  _       <- handler.onClose(connectionId,
-                                             WebSocketCloseEvent(e.code, e.reason, e.wasClean, errored)
-                             )
-                } yield ()
-              dispatcher.unsafeRunAndForget(close)
-            }
+          ws.onclose = { e: CloseEvent =>
+            val close: F[Unit] =
+              for {
+                _       <- s"WebSocket closed for URI [$uri]".traceF
+                errored <- isErrored.get
+                _       <- handler.onClose(connectionId,
+                                           WebSocketCloseEvent(e.code, e.reason, e.wasClean, errored)
+                           )
+              } yield ()
+            dispatcher.unsafeRunAndForget(close)
           }
         }
     } yield connection
 }
 
 object WebSocketJSBackend {
-  def apply[F[_]: Async: Logger]: WebSocketJSBackend[F] = new WebSocketJSBackend[F]
+  def apply[F[_]: Async: Logger](dispatcher: Dispatcher[F]): WebSocketJSBackend[F] =
```

Comment:
Will this be created on `explore` then?

Reply:
Yes. Apps will probably need to keep a global dispatcher, which can be passed here.